npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb

```diff
@@ -14,117 +14,95 @@
   end
```

Comment:
Ended up re-structuring this spec a fair bit as I found it hard to follow, de-nested some contexts and used existing projects and added a npm 7 workspace spec.